api/nodes/serializers.py

```diff
@@ -464,6 +467,19 @@ class NodeSerializer(TaxonomizableSerializerMixin, JSONAPISerializer):
     ))
 
     def get_current_user_permissions(self, obj):
+        request = self.context['request']
+        if StrictVersion(request.version) < StrictVersion('2.11'):
+            return self.get_current_user_permissions_legacy(obj)
+        user = self.context['request'].user
+        if isinstance(user, AnonymousUser):
+            perms = []
+        else:
+            perms = obj.get_permissions(user)
+            if user in obj.parent_admin_contributors:
```

Comment:
Calling `parent_admin_contributors` is fairly expensive, how about only doing it if perms comes up empty after `get_permissions`?